Lab-2/BinaryTree.h

```diff
@@ -0,0 +1,109 @@
+#pragma once
+#include <vector>
+#include <QWidget>
+
+class BinaryTree
+{
+public:
+    class Node;
+
+public:
+    BinaryTree() = default;
+    BinaryTree(const BinaryTree &other);
+
+    ~BinaryTree();
+
+    int getKey();
+
+    Node* root();
+
+    void clear();
+    static void clear(Node* root);
+
+    BinaryTree clone() const;
+    static BinaryTree clone(Node *root);
+    void operator=(const BinaryTree &other);
+
+    bool isIdeal() const;
+    bool isBalanced() const;
+
+    Node *addNode(int key);
+
+    void printHorizontal(int levelSpacing = 4) const;
+    void printHorizontal(Node *root, int marginLeft, int levelSpacing) const;
+
+    static void clearLeftChild(Node *root = nullptr);
+    static void clearRightChild(Node *root = nullptr);
+    static void clearChild(Node *root = nullptr);
+
+    bool isEmpty() const;
+
+    BinaryTree copyRightChild() const;
+    BinaryTree copyLeftChild() const;
+
+    int height() const;
+    int height(int) const;
+    int power() const;
+
+    int max() const;
+    int min() const;
+
+    bool remove(const int key);
+
+    BinaryTree find(const int key) const;
+
+    void printList() const;
+
+    operator std::vector<int>();
+private:
+    BinaryTree(Node* root);
```

Comment:
```suggestion
    explicit BinaryTree(Node* root);
```
Конструктор несколько спеифичен, лучше избежать неявных срабатываний.